dev/com.ibm.ws.cdi.internal/resources/com/ibm/ws/cdi/internal/resources/CDI.nlsprops

```diff
@@ -104,3 +104,7 @@ enableImplicitBeanArchives.conflict.CWOWB1017W.useraction=Update your server.xml
 unversioned.beansXML.warning.CWOWB1018W=CWOWB1018W: Using an unversioned CDI beans.xml file might lead to unexpected behaviour when using CDI 4.0 or later: {0}
 unversioned.beansXML.warning.CWOWB1018W.explanation=Prior to CDI 4.0, a non-empty but unversioned beans.xml file meant that the archive was treated as an explicit bean archive (all classes are eligible to be beans). In CDI 4.0 or later, the archive is taken to be an implicit bean archive (only annotated classes are eligible be beans).
 unversioned.beansXML.warning.CWOWB1018W.useraction=Update the beans.xml file within your application to use a newer version of the schema and ensure that the version and bean-discovery-mode attributes are properly set.
+
+MODULE_NOT_OPENED.CWOWB1019E=CWOWB1019E: Running on Java 17 without exposing the modules needed for CDI to create proxies.
```

Comment:
Would the same issue apply to >17 as well? e.g. 21, as the message currently implies that only 17 is affected and the behavior has changed back in later versions, which I assume it hasn't. So if someone on Java21 hit this while the error is confusing given they would likely know they are on 21 and we are seemingly stating they are on 17. So saying `17+` would allow for going forward.

For the explanation saying `Java 17 onwards,` instead of `Unlike previous versions of Java, Java 17` again supports multiple versions 

For the user action I think this can be simplified and more in line with existing messages such as 
https://github.com/OpenLiberty/open-liberty/blob/6e1dca820be198e36a4c7f3e01eee9a9851e653f/dev/com.ibm.ws.kernel.cmdline/resources/com/ibm/ws/kernel/boot/cmdline/resources/CommandLineMessages.nlsprops#L36
so 
`Set the JAVA_HOME environment variable to the path of a Java SDK, and then try the command again. If this is insufficient, add the following JVM argument: --add-opens=java.base/java.lang=ALL-UNNAMED`